### PR TITLE
[minor] add SearchbyParameter function

### PR DESCRIPTION
**Pull request by Eastwilding**
## 経緯
NGT Capi.h の ngt_search_index：https://github.com/yahoojapan/NGT/blob/master/lib/NGT/Capi.h#L86
に検索パラメーターを追加するため、
構造体と関数を追加しました：
https://github.com/yahoojapan/NGT/blob/master/lib/NGT/Capi.h#L39-#L44
https://github.com/yahoojapan/NGT/blob/master/lib/NGT/Capi.h#L88

## 修正内容
上記の変更を
gongtで利用するために、
対応した構造体と関数を追加します。

宜しくお願い致します。

### CODE REVIEW

File: gongt.go [hunk 1]
```diff
@@ -505,6 +517,78 @@ func (n *NGT) Search(vec []float64, size int, epsilon float64) ([]SearchResult,
 	return result[:idx], nil
 }
 
+// StrictSearchbyParameter is C type stricted search function
+func StrictSearchbyParameter(vec []float64, ssp StrictSearchParameter) ([]StrictSearchResult, error) {
```
Comment (kpango) [inline gongt.go]: by should be By

Comment (Eastwilding) [reply gongt.go]: thanks for review! 
replaced all "by" using "By"

**Comment by kpango** [comment]
It is not good to pass a struct, so we are discussing with the team, I think that it is better to think about minor fix in core lib so please give me more time to merge

**Comment by Eastwilding** [comment]
Totally agree! I didn't like to pass a struct either, but failed to find a better way. Pls take your time and I would like to reflect all the feedback from your team! Thanks

**Comment by kpango** [comment]
As a result of the discussion, it is better to implement additional arguments to core lib's search_index method, so we will close this PR.

**Pull request closed without merging**